path/path_test.go

```diff
@@ -28,3 +28,27 @@ func TestPathParsing(t *testing.T) {
 		}
 	}
 }
+
+func TestPathCleaning(t *testing.T) {
+	cases := map[string]string{
+		// .. should strip the preceding path segment
```

Comment:
Maybe this should be pushed to ipfs/go-ipld?



—
Sent from Mailbox

On Mon, Sep 21, 2015 at 6:16 PM, Will Glynn <notifications@github.com>
wrote:

>> @@ -28,3 +28,27 @@ func TestPathParsing(t *testing.T) {
>>  		}
>>  	}
>>  }
>> +
>> +func TestPathCleaning(t *testing.T) {
>> +	cases := map[string]string{
>> +		// .. should strip the preceding path segment
> This is the same logic as [POSIX pathname resolution](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_12) and [relative URI resolution](https://tools.ietf.org/html/rfc3986#section-5.2.4).
> User-specified objects named `.` and `..` would be un-addressable in most contexts already, and even if they were addressable in some contexts, it would be very confusing if they had a different meaning than in POSIX and in URIs.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/ipfs/go-ipfs/pull/1740/files#r40046047